Makefile.pre.in

```diff
@@ -2772,6 +2772,8 @@ funny:
 .PHONY: patchcheck
 patchcheck: all
 	$(RUNSHARED) ./$(BUILDPYTHON) $(srcdir)/Tools/patchcheck/patchcheck.py
+	@$(RUNSHARED) ./$(BUILDPYTHON) -m pre_commit --version >/dev/null 2>&1 || $(RUNSHARED) ./$(BUILDPYTHON) -m pip install pre-commit >/dev/null 2>&1
+	$(RUNSHARED) ./$(BUILDPYTHON) -m pre_commit run --all-files
```

Comment:
I'm thinking it would be a good idea to add a `make lint` target like we have for the PEPs repo (and Pillow), which seems to work there.

Perhaps we should add `make lint` now and have it run pre-commit; and maybe also patchcheck?

Reply:
I'd agree. We should also update the devguide (currently only [patchcheck is mentioned](https://devguide.python.org/getting-started/pull-request-lifecycle/#patchcheck))

Reply:
I've removed the `Makefile` changes from this PR, let's decide what to later after discussion:

https://discuss.python.org/t/do-you-use-make-patchcheck/34743?u=hugovk